packages/ide/src/upload.ts

```diff
@@ -133,20 +133,19 @@ export class Upload {
 	 * On failure, show the error and remove the directory from the queue.
 	 */
 	private async prepareDirectory(path: string, dir: IUploadableDirectory): Promise<void> {
-		await Promise.all([
-			promisify(exec)(`mkdir -p ${escapePath(path)}`).catch((error) => {
-				const message = error.message.toLowerCase();
-				if (message.includes("file exists")) {
-					throw new Error(`Unable to create directory at ${path} because a file exists there`);
-				}
-				throw new Error(error.message || `Unable to upload ${path}`);
-			}),
+		await promisify(exec)(`mkdir -p ${escapePath(path)}`).catch((error) => {
```

Comment:
I think this first catch block should be merged into the one below so we don't have two catches in our code.

Reply:
Done